pkg/util/mount/mount.go

```diff
@@ -166,19 +183,21 @@ func (mounter *SafeFormatAndMount) FormatAndMount(source string, target string,
 	return mounter.formatAndMount(source, target, fstype, options)
 }
 
-// GetMountRefsByDev finds all references to the device provided
-// by mountPath; returns a list of paths.
-func GetMountRefsByDev(mounter Interface, mountPath string) ([]string, error) {
+// GetMountRefsBySource finds all references to the mount source provided
+// by mountSource; returns a list of paths.
+// mountSource could be a mountpoint path, device or a normal directory (for
+// bind mount).
+func GetMountRefsBySource(mounter Interface, mountSource string) ([]string, error) {
 	mps, err := mounter.List()
```

Comment:
Would prefer having a new List method that uses mountinfo instead of changing existing List

Reply:
@msau42 Done in https://github.com/kubernetes/kubernetes/pull/62102/commits/e710fa51e1631090f9c0913765b58a94ee1f8e6a.

I don't find any downside to extend `MountPoint` struct to include information from `/proc/<pid>/mountinfo`, except it exposes more linux-specific details. However, I agree this is unnecessary to fix the issue. IMHO, maybe we can unify them after more tests on various platform.